### ISSUE TITLE
Dark Wizard - Freezing in US and JP versions

### ISSUE THREAD

**Issue by M-Walrus** [MiSTer-devel/MegaCD_MiSTer#25]
Dark Wizard US version (reDump set) - with any US bios I have tried, freezes randomly when exiting towns. 

The screen will show the background for the Sega CD bios upon exiting the village, and the system will freeze on that screen. This doesn't always happen, but it seems to be often. I had it happen in both the first and second village you go through.

-----

Dark Wizard - Yomigaerishi Yami no Madoushi (JP reDump set) - Freezes when exiting battle while "battle graphics" are enabled (top option in options menu)

When trying the JP version, the game froze when exiting battle on the first enemy I encountered. Using JP 2.00c bios.

**Comment by M-Walrus** [comment]
It seems the freezes are all over the place in this one , no matter what you do. 

Just tried with the latest 20.06.02 build and it froze on the name entry screen before the game even started.

**Comment by aberusugi** [comment]
What does it do when you change to a Mega-CD JP v1.00 BIOS?

**Comment by M-Walrus** [comment]
Bios switching seems to be broken. 

When i load a new bios (JP 1.00s), and then load a disc, it reverts back to my bootrom bios (US 2.0 region free) every time.

Freezes seem to happen at the same place in the JP version of the game. When confirming name entry at the beginning it froze on me immediately. This is with the US region free bios though, as the switching isnt working for me.

**Comment by sorgelig** [comment]
Every disk load also reloads the BIOS.
Actually i'm planning to remove manual BIOS load as it's not so compatible with new way to load the BIOS.
just place cd_bios.rom inside the game's folder if specific game requires a specific BIOS.

**Comment by aberusugi** [comment]
Did you take your jp bios 1.00 and rename it cd_bios.rom and place it in the directory of your game? Try to organize the ROMs by folder for megacd and put jp bios (renamed cd_bios.rom) in jp folders and us bios in us folders.

**Comment by sorgelig** [comment]
You have to place specific cd_bios.rom into CD's folder, so it will be loaded upon disk loading.

**Comment by M-Walrus** [comment]
Just wanted to update that this is still happening in the latest version of the core. 

It is much more infrequent now, but in a game like this if you forget to save, a freeze could set you back many hours of gameplay.

since it can be difficult to test a game like this, i'll provide a savegame and video instruction to reproduce the freezing, as I have noticed it is more common in some cities.

**Comment by birdybro** [comment]
What's the hash on your bios you are using? Mine is the one that is supposed to be used currently, renamed to boot.rom (nabbed from the update_all script):

CRC32: 7E86C5DA
MD5: 14DB9657BBAA6FBB9249752424DC0CE4

I haven't played into the game as much as you so I haven't seen this yet. This is the ROM I've been using:

![image](https://user-images.githubusercontent.com/16388068/94372756-ae6b0400-00bd-11eb-8727-2c2936cfd29d.png)
)

Is your situation the same as mine?

**Comment by birdybro** [comment]
The game locks up at the title screen in the current core. Testing a new build with the framework updated now to see if it's a marginal timing issue like other cores have seen problems with. Will update with findings.

**Comment by Zophar1** [comment]
The game is making it past the title with the latest June 2022 core, but I have not played it past the very beginning yet to know if it's still freezing.

**Comment by birdybro** [comment]
Yes, I think the freezes I noticed were due to marginal timing issues, similar issues that occurred in the Neo Geo core around that time.

**Comment by bomarmonk** [comment]
Loading Dark Wizard (redump) for mega drive core (with the latest update):  the game freezes for me before the title screen, with the “produced by or under license from sega” flashing green.

**Comment by czackitc** [comment]
On megacd core, exiting a town game reset game to the sega cd insert disc screen. Experienced 3 freezes since then, once in the options menu and twice after selecting leader and concluding the intro fmv.
